src/cargo/core/registry.rs

```diff
@@ -166,9 +166,9 @@ impl<'cfg> PackageRegistry<'cfg> {
 
     pub fn register_lock(&mut self, id: PackageId, deps: Vec<PackageId>) {
         let sub_map = self.locked.entry(id.source_id().clone())
-                                 .or_insert(HashMap::new());
+                                 .or_insert_with(HashMap::new);
```

Comment:
Yes but `or_insert_with(HashMap::new)` in this case looks less readable to me, so I'd prefer to leave as-is.